src/SDKs/Batch/DataPlane/Azure.Batch/CloudJob.cs

```diff
@@ -573,7 +573,39 @@ public void Refresh(DetailLevel detailLevel = null, IEnumerable<BatchClientBehav
             asyncTask.WaitAndUnaggregateException(this.CustomBehaviors, additionalBehaviors);
         }
 
-#endregion // CloudJob
+        /// <summary>
+        /// Gets the task counts for the job.
+        /// </summary>
+        /// <param name="detailLevel">A <see cref="DetailLevel"/> used for controlling which properties are retrieved from the service.</param>
+        /// <param name="additionalBehaviors">A collection of <see cref="BatchClientBehavior"/> instances that are applied to the Batch service request after the <see cref="CustomBehaviors"/> and <paramref name="detailLevel"/>.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> for controlling the lifetime of the asynchronous operation.</param>
+        /// <returns>A <see cref="TaskCounts"/> object containing the task counts for the job.</returns>
```

Comment:
You're correct: I recall now that we wanted to direct people from the sync methods towards the async ones but not in the other direction.  We _do_, however, usually put a remark "The noodling operation runs asynchronously." - suggest we add one of those for now, and if we feel they are no longer worthwhile then ditch them project-wide in a separate PR.